src/maths.rs

```diff
@@ -123,6 +127,15 @@ impl MathematicalOps for Decimal {
 
         result
     }
+    
+    fn checked_exp(&self) -> Option<Decimal> {
+        // Current calculation method will overflow when e is raised to a power larger than around 11.7578.
+        if self > Decimal::from_parts(117578, 0, 0, false, 4) {
```

Comment:
Hmm, I'm not sure if this is the approach I'd go. While it's a nice sanity check, the code as you currently have it still has risk of `panic`ing which is not what I'd expect for a `checked_` function. 
It'd be more efficient and effective to modify `exp_with_tolerance` to have a checked variant. I suspect overflow either occurs on one of the lines:
```rust
term = self * term;
let next = result + (term / factorial);
```
My gut, is that it's on the multiplication. i.e. if you modify the code to use `checked_mul` and handle the `Option` result. You could in fact use the internal `ops` function if you chose to, handling the `Result` directly.

The other thing I'd request is a test for this function. Something to show it working with `Some` value and also safely handling an overflow by returning `None`.

Reply:
That's right on.  The overflow is on the multiply.

Reply:
I'll try your suggestion.